test/built-ins/GeneratorPrototype/return/try-finally-set-property-within-try.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2022 Bo Pang. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 25.3.1.3
+description: >
+    When a generator is paused within a `try` block of a `try..finally`
+    statement, `return` should interrupt control flow as if a `return`
+    statement had appeared at that location in the function body.
+    And the `finally` block runs first.
+features: [generators]
+---*/
+
+
+var obj={foo:undefined};
+function* g() {
+  try {
+    obj.foo=yield;
+  } finally {
+    return 1;
+  }
+}
+var iter = g();
+var result;
+
+iter.next();
+let ret=iter.return(45);
+assert.sameValue(obj.foo,undefined,'`obj.foo` did not set');
+assert.sameValue(ret,1,'return in finally executes first');
```

Comment:
This assertion is [failing](https://app.circleci.com/pipelines/github/tc39/test262/4541/workflows/73bdd566-06d5-4713-8a55-5837484b80a8/jobs/42468), because the shape of the result of `iter.return(45)` is not a primitive but rather an [_IteratorResult_ object](https://tc39.es/ecma262/multipage/control-abstraction-objects.html#sec-iteratorresult-interface) with `value` and `done` properties per [_Iterator_ Interface Optional Properties](https://tc39.es/ecma262/multipage/control-abstraction-objects.html#table-iterator-interface-optional-properties).

Reply:
Oh sorry, I forgot to write that.Thanks a lot!